src/go/MONIT/monit.go

```diff
@@ -60,6 +60,87 @@ func (c *StompConfig) String() string {
 	return fmt.Sprintf("<StompConfig uri=%s producer=%s topic=%s>", c.URI, c.Producer, c.Topic)
 }
 
+// represents parameters needed to run queries
+type Params struct {
+	URL       string
+	DBName    string
+	DBID      int
+	Database  string
+	DBType    string
+	ESAPI     string
+	ShortDBID int    // index id in a short term storage cluster
+	LongDBID  int    // index id in a long term storage cluster
+	ShortURL  string // request URL for short term cluster stats
+	LongURL   string // request URL for long term cluster stats
+	Token     string
+	Query     string
+	Verbose   int
+	Config    StompConfig
+	Inject    bool
+	HDFS      string
+}
+
+// builder function to get default params for stats query
+func NewStatsParams(url string, token string, q string, verbose int, config StompConfig, inject bool, hdfs string) *Params {
+	// per our discussion with MONIT team (RQF1647972)
+	// we only need to use some dbid for _stats ES API, here I use
+	// 9573, the id of monit_prod_cms-es-size_raw_elasticsearch datasource, and 9582
+	// the id of monit_prod_condor_agg_metric datasource
+	return &Params{
+		ShortDBID: 9573,
+		LongDBID:  9582,
+		ShortURL:  fmt.Sprintf("%s/api/datasources/proxy/9573/monit_prod*/_stats/store", url),
+		LongURL:   fmt.Sprintf("%s/api/datasources/proxy/9582/monit_prod*/_stats/store", url),
+		Token:     token,
+		Query:     q,
+		Verbose:   verbose,
+		Config:    config,
+		Inject:    inject,
+		HDFS:      hdfs,
+	}
+}
+
+// builder function of params for queries
+func NewParams(url string, token string, q string, dbname string, dbid int, database string, dbtype string, esapi string, hdfs string, verbose int) *Params {
+	return &Params{
```

Comment:
For every parameter of the struct please provide proper validation of data-type, see how it is done in dbs2go code https://github.com/dmwm/dbs2go/blob/master/dbs/datasets.go#L284 and read about Go validator in here https://github.com/go-playground/validator